### PR TITLE
vectorMap: change demo Dynamic Viewport adding an option for control bar

### PR DESCRIPTION
**Pull request by EvgeniiMorozovDevSoft**

### CODE REVIEW

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 1]
```diff
@@ -6,31 +6,48 @@
...
+    <div class="column">
+      <div class="option">
+        <span>Continent</span>
+        <dx-select-box
+          [dataSource]="viewportData"
+          [width]="210"
+          displayExpr="continent"
+          valueExpr="coordinates"
+          [value]="viewportData[0].coordinates"
+          (onValueChanged)="onValueChanged($event)"
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html:24]: onValueChanged is needed?

File: JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.js [hunk 2]
```diff
@@ -15,7 +15,20 @@ window.onload = function () {
...
+        zoomVisible: true,
+      },
+    },
+    switchPan: {
+      value: true,
+      onValueChanged: (data) => $('#vector-map').dxVectorMap('instance').option('controlBar.panVisible', data.value),
     },
+    switchZoom: {
+      value: true,
+      onValueChanged: (data) => $('#vector-map').dxVectorMap('instance').option('controlBar.zoomVisible', data.value),
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.js]: do not needed use this approach (with onValueChanged event using).

File: JSDemos/Demos/VectorMap/DynamicViewport/jQuery/index.js [hunk 3]
```diff
@@ -13,6 +13,17 @@ $(() => {
     },
   }).dxVectorMap('instance');
 
+  $('#switchPan').dxSwitch({
+    value: true,
+    onValueChanged: (e) => map.option('controlBar.panVisible', e.value),
+
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/VectorMap/DynamicViewport/jQuery/index.js]: extra line

**Review by alexanderPolosatov — CHANGES_REQUESTED** [review]

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 4]
```diff
@@ -6,31 +6,48 @@
   (onCenterChanged)="centerChanged($event)"
 >
   <dxi-layer [dataSource]="worldMap"></dxi-layer>
+  <dxo-control-bar [panVisible]="pan.value" [zoomVisible]="zoom.value">
+    ></dxo-control-bar
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html]: delete unnecessary `>` and make <dxo-control-bar></dxo-control-bar> on one line

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 5]
```diff
@@ -6,31 +6,48 @@
...
+        <dx-text-box
+          [value]="centerValue"
+          [readOnly]="true"
+          [width]="210"
+        ></dx-text-box>
+      </div>
+    </div>
+    <div class="column">
+      <div class="option">
+        <span>pan Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html]: Check text from correctors

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 6]
```diff
@@ -6,31 +6,48 @@
...
+        <dx-switch [value]="panVisible" #pan></dx-switch>
+      </div>
+      <div class="option">
+        <span>zoom Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html]: Check text from correctors

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 7]
```diff
@@ -6,31 +6,48 @@
...
+          [value]="centerValue"
+          [readOnly]="true"
+          [width]="210"
+        ></dx-text-box>
+      </div>
+    </div>
+    <div class="column">
+      <div class="option">
+        <span>pan Visible</span>
+        <dx-switch [value]="panVisible" #pan></dx-switch>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html]: #pan - is this needed?

File: JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html [hunk 8]
```diff
@@ -6,31 +6,48 @@
...
+      </div>
+      <div class="option">
+        <span>zoom Visible</span>
+        <dx-switch [value]="zoomVisible" #zoom></dx-switch>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Angular/app/app.component.html]: #zoom - is this needed?

File: JSDemos/Demos/VectorMap/DynamicViewport/AngularJS/index.html [hunk 9]
```diff
@@ -25,17 +25,31 @@
...
+              <div dx-text-box="zoomFactor"></div>
+            </div>
+            <div class="option">
+              <span>Center</span>
+              <div dx-text-box="center"></div>
+            </div>
+          </div>
+          <div class="column">
+            <div class="option">
+              <span>pan Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/AngularJS/index.html]: Check text

File: JSDemos/Demos/VectorMap/DynamicViewport/AngularJS/index.html [hunk 10]
```diff
@@ -25,17 +25,31 @@
...
+              <div dx-switch="handlePanVisibleSwitch"></div>
+            </div>
+            <div class="option">
+              <span>zoom Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/AngularJS/index.html]: check text

File: JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html [hunk 11]
```diff
@@ -26,17 +26,31 @@
...
+                            <div data-bind="dxTextBox: zoomFactor"></div>
+                        </div>
+                        <div class="option">
+                            <span>Center</span>
+                            <div data-bind="dxTextBox: center"></div>
+                        </div>
+                    </div>
+                    <div class="column">
+                        <div class="option">
+                            <span>pan Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html]: change span

File: JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html [hunk 12]
```diff
@@ -26,17 +26,31 @@
...
+                            <div data-bind="dxTextBox: center"></div>
+                        </div>
+                    </div>
+                    <div class="column">
+                        <div class="option">
+                            <span>pan Visible</span> 
+                            <div data-bind="dxSwitch: chooseContinent"></div>
+                        </div>
+                        <div class="option">
+                            <span>zoom Visible</span>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html]: change span

File: JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html [hunk 13]
```diff
@@ -26,17 +26,31 @@
...
+                        </div>
+                        <div class="option">
+                            <span>Center</span>
+                            <div data-bind="dxTextBox: center"></div>
+                        </div>
+                    </div>
+                    <div class="column">
+                        <div class="option">
+                            <span>pan Visible</span> 
+                            <div data-bind="dxSwitch: chooseContinent"></div>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html]: chooseContinent for panVisible?

File: JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html [hunk 14]
```diff
@@ -26,17 +26,31 @@
...
+                        </div>
+                        <div class="option">
+                            <span>zoom Visible</span> 
+                            <div data-bind="dxSwitch: chooseContinent"></div>
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/Knockout/index.html]: chooseContinent for zoomVisible?

File: MVCDemos/Views/VectorMap/DynamicViewport.cshtml [hunk 15]
```diff
@@ -9,35 +9,57 @@
...
+                    .ReadOnly(true)
+                    .Value("0.000, 46.036")
+                )
+            </div>
+        </div>
+        <div class="column">
+             <div class="option">
+                <span>pan Visible</span>
+               @(Html.DevExtreme().Switch()
+               .ID("pan")
```
Comment (Krijovnick) [inline MVCDemos/Views/VectorMap/DynamicViewport.cshtml]: Is ID needed?

File: NetCoreDemos/Views/VectorMap/DynamicViewport.cshtml [hunk 16]
```diff
@@ -1,4 +1,4 @@
-﻿@model IEnumerable<DevExtreme.NETCore.Demos.Models.ViewportCoordinate>
+﻿@model IEnumerable<DevExtreme.MVC.Demos.Models.ViewportCoordinate>
```
Comment (Krijovnick) [inline NetCoreDemos/Views/VectorMap/DynamicViewport.cshtml]: Why NetCore was changed to MVC?

**Review by Krijovnick — CHANGES_REQUESTED** [review]

File: JSDemos/Demos/VectorMap/DynamicViewport/React/App.js [hunk 17]
```diff
@@ -53,33 +65,54 @@ class App extends React.Component {
...
+                  readOnly={true}
+                />
+              </div>
+            </div>
+            <div className="column">
+              <div className="option">
+                <span>pan Visible</span>&nbsp;
+                <Switch
+                  value={this.state.panVisible}
+                  onValueChanged={this.panVisibleChanged} />
```
Comment (Krijovnick) [inline JSDemos/Demos/VectorMap/DynamicViewport/React/App.js]: In react should uses onValueChange option - https://js.devexpress.com/Documentation/Guide/React_Components/State_Management/?search=onValueChange=

**Review by alexanderPolosatov — DISMISSED** [review]

**Review by alexanderPolosatov — DISMISSED** [review]

**Review by Krijovnick — DISMISSED** [review]

**Review by Krijovnick — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**